sympy/polys/fields.py

```diff
@@ -190,11 +190,17 @@ def field_new(self, element):
         if isinstance(element, FracElement):
             if self == element.field:
                 return element
+            elif self.domain.field == element.field:
+                return self.ground_new(element)
             else:
                 raise NotImplementedError("conversion")
         elif isinstance(element, PolyElement):
             denom, numer = element.clear_denoms()
-            numer = numer.set_ring(self.ring)
+            if isinstance(self.domain, PolynomialRing) and \
+                numer.ring == self.domain.ring:
+                numer = self.ring.ground_new(numer)
+            else:
```

Comment:
What could be done if `self.domain` is a `FractionField`?

Reply:
I've fixed them